### PR TITLE
Berkley 1.5.3 flr 622

### PR DESCRIPTION
**Pull request by ligadata-saleh**
changed for both h2db and sql server storage adapters  , use the 

./GenerateKeys.sh --algorithm RSA --generateSampleKeys

then 

./GenerateKeys.sh --algorithm RSA --publicKeyFile /tmp/keys/public.key --privateKeyFile /tmp/keys/private.key --password test

after that change the config files  

cluster config :
    {
      "ClusterId": "ligadata1",
      "SystemCatalog": {
        "StoreType": "h2db",
        "connectionMode": "embedded",
        "SchemaName": "kamanja",
        "Location": "/Users/saleh/Desktop/Kamanja-1.5.3_2.11/storage/syscatalog",
        "portnumber": "9100",
        "algorithm": "RSA",
        "privatekey": "/tmp/keys/private.key",
        "user": "test",
        "password": "PPItCu93DI3UcKtIbykPzt5YSzElFPqTda66vLDJmpMogphuFeFg6HkEtNJdiH1aZ2nuC7HxEIC1q6SgPELL1rr9diPnl9Bxi6RDTmgL2yUfW7Oil5abcSpHus9J7Ld1pGCVfGKiHXXqZXfvSzRAh+TIPhiUZIl91V/0M0AHq/M="
      }

metadata :
MetadataDataStore={"StoreType": "h2db","connectionMode": "embedded","SchemaName": "kamanja","Location": "/Users/saleh/Desktop/Kamanja-1.5.3_2.11/storage","portnumber": "9100","algorithm": "RSA","privatekey": "/tmp/keys/private.key","user": "test","password": "PPItCu93DI3UcKtIbykPzt5YSzElFPqTda66vLDJmpMogphuFeFg6HkEtNJdiH1aZ2nuC7HxEIC1q6SgPELL1rr9diPnl9Bxi6RDTmgL2yUfW7Oil5abcSpHus9J7Ld1pGCVfGKiHXXqZXfvSzRAh+TIPhiUZIl91V/0M0AHq/M="}

engin config : 
{"StoreType": "h2db","connectionMode": "embedded","SchemaName": "kamanja","Location": "/Users/saleh/Desktop/Kamanja-1.5.3_2.11/storage","portnumber": "9100","algorithm": "RSA","privatekey": "/tmp/keys/private.key","user": "test","password": "PPItCu93DI3UcKtIbykPzt5YSzElFPqTda66vLDJmpMogphuFeFg6HkEtNJdiH1aZ2nuC7HxEIC1q6SgPELL1rr9diPnl9Bxi6RDTmgL2yUfW7Oil5abcSpHus9J7Ld1pGCVfGKiHXXqZXfvSzRAh+TIPhiUZIl91V/0M0AHq/M="}

### CODE REVIEW

**Comment by ligadata-pokuri** [comment]
@ligadata-william, can you review the code and test this?

File: trunk/Storage/SqlServer/src/main/scala/com/ligadata/keyvaluestore/SqlServerAdapter.scala [hunk 1]
```diff
@@ -182,9 +183,27 @@ class SqlServerAdapter(val kvManagerLoader: KamanjaLoaderInfo, val datastoreConf
     SchemaName = user
   }
 
+  var algorithm = None: Option[String];
+  if (parsed_json.contains("algorithm")) {
+    algorithm = Some(parsed_json.get("algorithm").get.toString.trim)
+  }
+
```
Comment (ligadata-william) [inline trunk/Storage/SqlServer/src/main/scala/com/ligadata/keyvaluestore/SqlServerAdapter.scala]: I would change this to:
```
val algorithm: Option[String] = if(parsed_json.contains("algorithm") Some(parsed_json("algorithm").toString.trim else None
```
This keeps it a one-liner and you avoid having a var that may accidentally be changed in future code changes.

File: trunk/Storage/SqlServer/src/main/scala/com/ligadata/keyvaluestore/SqlServerAdapter.scala [hunk 2]
```diff
@@ -182,9 +183,27 @@ class SqlServerAdapter(val kvManagerLoader: KamanjaLoaderInfo, val datastoreConf
...
+  var privatekey = None: Option[String];
+  if (parsed_json.contains("privatekey")) {
+    privatekey = Some(parsed_json.get("privatekey").get.toString.trim)
+  }
+
```
Comment (ligadata-william) [inline trunk/Storage/SqlServer/src/main/scala/com/ligadata/keyvaluestore/SqlServerAdapter.scala]: I would change this to:
```
val privatekey: Option[String] = if(parsed_json.contains("privatekey") Some(parsed_json("privatekey").toString.trim else None
```
This keeps it a one-liner and you avoid having a var that may accidentally be changed in future code changes.

**Comment by ligadata-saleh** [comment]
change private key and algorithm to val as @ligadata-william suggested

**Comment by ligadata-william** [comment]
This has been merged into Berkley-1.6.2. Closing.

**Pull request closed without merging**